### ISSUE TITLE
HID on Windows

### ISSUE THREAD

**Issue by MOAMaster** [chessnutech/EasyLinkSDK#2]
I opened hidapi.sln in Vis Studio 2019. The DLL and the EXE successfully built.

But, when I try to execute the EXE, I get a notice that hidapi.dll is not a valid Win32 application.

Not sure what to do next.
Any advice is appreciated.

**Comment by chessnutech** [comment]
Hi chessnutech/EasyLinkSDK,  Thank you for reaching out.  Regarding your query, you only need to follow the steps outlined in the readme file and do not need to perform the actions mentioned in the other email. Please refer to the image below for guidance.  I hope our response helps. We await your reply.     Best,  Katherine/Team Chessnut     
                           
                            On
                            Wed, 3 Jul at 11:46 PM
                            ,  chessnutech/EasyLinkSDK ***@***.***>  wrote:
                             
 I opened hidapi.sln in Vis Studio 2019. The DLL and the EXE successfully built. 
 But, when I try to execute the EXE, I get a notice that hidapi.dll is not a valid Win32 application. 
 Not sure what to do next. 
Any advice is appreciated.

**Comment by MOAMaster** [comment]
OK, I am using Chessnut Air.

 

I was able to get the SDK to build.

I can connect to the board and get the version.

Nothing is returned when I ask for mcu version and ble version.

 

No problem with that.


But, I put a piece on the board.

Registered a callback for movement.

Went into realtime mode.

In otherwords, I implemented the sample program.

 

I initialize the board to start a game.

I move the piece, and do not get a report.


So, I have a couple of questions.

Does the SDK connect use USB, BLE, or either?

 

Must ALL the pieces be on the board for a movement

to be reported? Or an I have one piece, initiate the game,

then move a piece?

 

Thanks.

Bruce

**Comment by chessnutech** [comment]
Hi chessnutech/EasyLinkSDK,  Thank you for reaching out again.  Regarding your questions, here are our responses: 1. USB and BLE cannot be used simultaneously as it may cause data confusion and result in unknown issues. 2. When using HID, you can recognize multiple pieces and move multiple pieces simultaneously.  We hope our response helps you. This ticket will be closed and will no longer support replies.  If you have any further questions, please contact us by creating a new ticket.  Thank you for your understanding and support.     Best,  Katherine/Team Chessnut      
                           
                            On
                            Fri, 5 Jul at  2:33 AM
                            ,  chessnutech/EasyLinkSDK ***@***.***>  wrote:
                             
OK, I am using Chessnut Air.
 

 
 
 

 
I was able to get the SDK to build.
 

 
I can connect to the board and get the version.
 

 
Nothing is returned when I ask for mcu version and ble version.
 

 
 
 

 
No problem with that.
 

 

 
But, I put a piece on the board.
 

 
Registered a callback for movement.
 

 
Went into realtime mode.
 

 
In otherwords, I implemented the sample program.
 

 
 
 

 
I initialize the board to start a game.
 

 
I move the piece, and do not get a report.
 

 

 
So, I have a couple of questions.
 

 
Does the SDK connect use USB, BLE, or either?
 

 
 
 

 
Must ALL the pieces be on the board for a movement
 

 
to be reported? Or an I have one piece, initiate the game,
 

 
then move a piece?
 

 
 
 

 
Thanks.
 

 
Bruce
 

 
 
 

 
From: chessnutech ***@***.***> 
 
Sent: Thursday, July 4, 2024 12:53 AM
 
To: chessnutech/EasyLinkSDK ***@***.***>
 
Cc: MOAMaster ***@***.***>; Author ***@***.***>
 
Subject: Re: [chessnutech/EasyLinkSDK] HID on Windows (Issue #2)
 

 
 
 

 
Hi chessnutech/EasyLinkSDK, Thank you for reaching out. Regarding your query, you only need to follow the steps outlined in the readme file and do not need to perform the actions mentioned in the other email. Please refer to the image below for guidance. I hope our response helps. We await your reply. Best, Katherine/Team Chessnut 
 

 
On 
 
Wed, 3 Jul at 11:46 PM 
 
, chessnutech/EasyLinkSDK ***@***.*** <mailto:***@***.***> > wrote: 
 

 
I opened hidapi.sln in Vis Studio 2019. The DLL and the EXE successfully built. 
 
But, when I try to execute the EXE, I get a notice that hidapi.dll is not a valid Win32 application. 
 
Not sure what to do next. 
 
Any advice is appreciated.

**Comment by chessnutech** [comment]
I'm very sorry, there are some problems with the email replies, I deleted them and replied to you again

```
git clone ...
cd  EasyLinkSDK && mkdir build && cd build
cmake ..
cmake --build .
```

After following the above commands, you can successfully compile the dynamic library and static library files. If you need to run specific code, you need to write a separate program to call this dynamic library or static library

I will add a sample code later to show how to run it directly

**Comment by MOAMaster** [comment]
Thanks for following up,

but you don’t need to do this.

 

I ported the HIDAPI and the EasyLink SDK into visual studio,

wrote a layer for calling the Easylink SDK, an application driver,

and setup a link so I can call from java.

 

My app needs to be able to access the HD API for two machines

simultaneously. EasyLink assumes a single machine. I decided

to write a scaled down version SDK, much simpler that communicates

with multiple boards simultaneously. I have that working.

Am now working on code to determine the actual move based

on successive board dumps. Close to getting that working.

 

Also, I am integrating with a chess type app written in Java,

so I have written and tested java code that goes through

both EasyLink and my SDK to get to multiple machines

operating in parallel.


But, thanks for following through.


Bruce

**Comment by chessnutech** [comment]
Enjoy yourself

**Comment by MOAMaster** [comment]
That was not meant as a put down.

 

My version is not to the production quality as the easylink sdk.

In the easylink, you have a lot of needed complexity to handle

a wide range of situations. I would trust the Easylink sdk

if I were going to build a production quality app.

 

My requirements are very limited. So, I can afford to scale down

my version of the SDK to accommodate those limited requirements.

My SDK is NOT for commercial use. I ignore a lot of stuff that I don’t 

need to deal with in my application. And, I would not promote

it for commercial use. 

 

And, I would not have been able to create my scaled version

without having the full version to study.

 

Thanks for providing your version of the SDK as open source.

I will be doing the same for those who will be using my app.

 

Bruce
